engine/src/CalciteInterpreter.cpp

```diff
@@ -665,9 +703,15 @@ blazing_frame evaluate_split_query(std::vector<ral::io::data_loader> input_loade
 
 
 			// EnumerableTableScan(table=[[hr, joiner]])
-			scan_frame.add_table(input_table);
+			// TODO percy cudf0.12 could be some memory issues here if input_tables was moved before
+			for (int i = 0; i < input_table.size(); ++i) {
+				scan_frame_columns.push_back(std::move(input_table[i]));
```

Comment:
What about gdf inputs. You cant move them.

Reply:
same